diff --git a/days/03/03.bf b/days/03/03.bf
index 4d27cb3..faed79e 100644
--- a/days/03/03.bf
+++ b/days/03/03.bf
@@ -1,6 +1,6 @@
 Set up constants and loop counters
-1: 17 2: 19 3: 46 4: 31
-+++++++[>++>+++>+++++++>+++++<<<<-]>+++>-->--->----
+1: 7 2: 23 3: 46 4: 31
++++++++[>+>+++>+++++++>+++++<<<<-]>>++>--->----
 Variables:
 0: output count 1: loop counter 1 2: loop counter 2 5: current index 6: current input
 outer loop
@@ -13,9 +13,10 @@ outer loop
         skip index amount of chars
         <[<,>-]
         read char
-        <,.
+        <,
         skip rest of row
-        >+[[-],----------]++++++++++.[-]
+        >+[[-],#----------]
+        +[[-],#----------]
         copy 46 to pos7 for equality check
         <<<<[>>>>+>+<<<<<-]
         >>>>>[<<<<<+>>>>>-]
@@ -26,7 +27,7 @@ outer loop
         reset input
         [-]
         calculate new index
-        <+++
+        <+
         move index to pos7
         [>>+<<-]
         copy 31 to pos8
@@ -40,8 +41,8 @@ outer loop
         <[-]
     end of inner loop
     <<<<<<-]
-    reset inner loop counter to 19
-    >>>>>+++[<<<<<+++++++>>>>>-]<<<<<--
+    reset inner loop counter to 23
+    >>>>>+++[<<<<<+++++++>>>>>-]<<<<<++
 end of outer loop
 <-]
 move output to pos6
